### PR TITLE
Validating

### PR DESCRIPTION
**Pull request by mmulich**
Validate the content of a litezip. 

Depends on #2

### CODE REVIEW

**Comment by mmulich** [comment]
@Stackblocks please have a look at the completezip to litezip conversion. The following are setup instructions:

``` sh
git clone -b converting https://github.com/Connexions/cnx-litezip.git
cd cnx-litezip
# use a virtualenv of your choosing
python setup.py develop
python setup.py test
```

Using the existing completezip to litezip interface bits:

``` sh
# Download a completezip and unpack it.
# Something like:
http -d http://legacy.cnx.org/content/col11405/latest/complete
unzip col11405_1.2_complete.zip
# Then call the new utility on it:
completezip2litezip ./col11405_1.2_complete
```

Here is the new interface for validating the litezip:

``` sh
validate-litezip --help
# Using the converted results of the completezip2litezip mentioned above...
validate-litezip -v ./col11405_1.2_complete
echo $?  # should be 0
```

That likely won't produce anything since the content should be valid. To see some validation messages, have a look at:

``` sh
validate-litezip -v litezip/tests/data/invalid_litezip/
echo $?  # should be 1
```

The validation script here is more or less a toy deliverable, but it should show the results of the work thus far. It validates the identifiers are ok (e.g. `col11405` or `m40646`). And more importantly, it validates the contents of the files. Feel free to muck up the content of an actual litezip and see the results. Maybe try deleting a chunk of a document to see what happens. Good luck. 😃

**Comment by Stackblocks** [comment]
Ran into an issue where the test steps provided above have `git clone -b converting` but `git clone -b validating` was needed. Learned how to identify and select branch when cloning a repo, so turned out to be a good thing. Everything ran as advertised with test data. Pass.

**Comment by reedstrm** [comment]
@Stackblocks Did you try editing any of the cnxml yourself, to make it invalid? That'd be an interesting case ...

**Comment by Stackblocks** [comment]
@reedstrm I had not previously, so I mangled one of the files for completion's sake.

So after stripping out the first two lines of one of the CNXML files you get this response:

```
(cnx-litezip) Timber-Maniacs:cnx-litezip openstax$ validate-litezip -v ./col11405_1.2_complete
INFO : col11405_1.2_complete/m37151/index.cnxml: 3:8 -- error: unknown element "title"
INFO : col11405_1.2_complete/m37151/index.cnxml: 4:2 -- fatal: The markup in the document following the root element must be well-formed.

```

And if you were to strip out part of the Metadata lines you get the following:

```
(cnx-litezip) Timber-Maniacs:cnx-litezip openstax$ validate-litezip -v ./col11405_1.2_complete
INFO : col11405_1.2_complete/m37151/index.cnxml: 50:12 -- error: unfinished element
```

and a typo in one of the tags gets you:

```
(cnx-litezip) Timber-Maniacs:cnx-litezip openstax$ validate-litezip -v ./col11405_1.2_complete
INFO : col11405_1.2_complete/m37151/index.cnxml: 18:20 -- error: unknown element "firsname" from namespace "http://cnx.rice.edu/mdml"
INFO : col11405_1.2_complete/m37151/index.cnxml: 18:29 -- fatal: The element type "md:firsname" must be terminated by the matching end-tag "</md:firsname>".
```

File: litezip/validate.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+import re
+
+
+__all__ = (
+    'is_valid_identifier',
+)
+
+
+VALID_ID_REGEX = re.compile("^(col|m)\d{5}$")
```
Comment (reedstrm) [inline litezip/validate.py]: this should be `^(col\d{5,}\d*|m\d{4,})$` The first module is `m0001` (but the first collection is `col10040` for slightly historical reasons) and the commas to allow for more, in the future.

File: litezip/cli.py [hunk 2]
```diff
@@ -21,17 +31,59 @@ def _arg_parser():
     return parser
 
 
+console_logging_config = {
```
Comment (reedstrm) [inline litezip/cli.py]: We need to make this level of config and logging standard for all our cli tools.

**Review by reedstrm — APPROVED** [review]

**Review by reedstrm — CHANGES_REQUESTED** [review]
need to address the validation regex - see 'outdated' comment above.

File: litezip/validate.py (hunk 1 above)
Comment (reedstrm) [reply litezip/validate.py]: This is marked outdated, it is not.

**Comment by mmulich** [comment]
@reedstrm the regex has been updated. Please have another look.

**Review by reedstrm — APPROVED** [review]

**Pull request merged**